### PR TITLE
(GH-1850) Make examples section in docs

### LINKED ISSUE #1850: Expand the bolt example gallery
## Describe the Change You Would Like

How can we expand the bolt example gallery so that users have a better place to go to find examples for frequently asked questions.

- Update it as part of helping users in slack?
- Update it based on extracting examples from a complex module like enterprise_tasks?

## Immediate work

Extract 5ish examples from the enterprise tasks repo showing common bolt patterns into a new repo "bolt-examples"

### PR DESCRIPTION
**Pull request by lucywyman**
This adds a new 'examples' section to the docs, below 'Using Bolt with
Puppet in the site tree. It moves the 'automating windows targets' and
'tig' example docs to be in this section, and provides a brief
description of what each example exemplifies on the main examples page.
This provides a central place for users to look for examples, and
clarifies the site tree.

Comment for reviewers: My apologies that this diff is so annoying, the only change I made to the Automating Windows Targets doc was to add a more specific description before jumping right into the examples (and removing what was previously the intro text for the Examples section).

Closes #1850

!no-release-note

### CODE REVIEW

File: documentation/automating_windows_targets.md [hunk 1]
```diff
@@ -0,0 +1,433 @@
...
+
+### 1. Build a project-specific configuration
+
+Bolt runs in the context of a [Bolt project
+directory](bolt_project_directories.md). This directory contains all of the
+configuration, code, and data loaded by Bolt.
+
+1. Create a module called `bolt_choco_example`
+
+```
```
Comment (hestonhoffman) [inline documentation/automating_windows_targets.md:201]: From this point of the doc onwards, the ordered lists go all funky. I think the content between each list item needs to be indented. I can fix this in a separate ticket if you like. This doc needs updates anyway.

**Comment by hestonhoffman** [comment]
I think we should add links to the examples doc and examples repo under "Learn the basics" in bolt.md. Something like:
```
Bolt examples: Learn about how Bolt can help you automate some common tasks.
Bolt example repo: A curated repo of common Bolt patterns that you can use as a reference point.  
```
I think we can also remove that link to the online learning, since it's so out of date at this point. 

(bolt.md is a little annoying to edit cos it's mostly in raw HTML 😬 )

File: documentation/bolt_examples.md [hunk 2]
```diff
@@ -1,440 +1,13 @@
...
-Bolt lets you automate almost any task you can think of. These are some of the
-common scenarios we've come across.
+Bolt lets you automate almost any task you can think of. These examples walk you through beginning
+and intermediate level Bolt use cases, demonstrating Bolt concepts along the way. You can find
+shorter examples of common Bolt patterns in the [Bolt examples
+repo](https://github.com/puppetlabs/bolt-examples), which is more of a reference than a learning
+tool.
 
 If you'd like to share a real-world use case, reach out to us in the #bolt
 channel on [Slack](https://slack.puppet.com).
```
Comment (hestonhoffman) [inline documentation/bolt_examples.md:10]: I like this! Let's add a similar blurb to the bottom of each page:
```
Do you have a real-world use case for Bolt that you'd like to share? Reach out to us in the #bolt
channel on [Slack](https://slack.puppet.com).
```

**Comment by lucywyman** [comment]
No worries on the bolt.md - I cut my teeth writing raw HTML and CSS to customize my Neopet pet pages.

**Comment by hestonhoffman** [comment]
Haha, nice!

File: documentation/bolt.md [hunk 3]
```diff
@@ -54,7 +54,8 @@ you are not required to install any agent software.
     <p>Learn the basics
         <ul>
             <li><a class="xref" href="getting_started_with_bolt.md">Getting started with Bolt</a></li>
-            <li><a class="xref" target="_blank" href="https://learn.puppet.com/course/puppet-orchestration-bolt-and-tasks?_ga=2.158319738.1526297716.1533055277-261802629.1531434605">Online training</a> - Learn how to use Bolt to run a command, a script, and a task.</li>
+            <li><a class="xref" href="bolt_examples.md">Bolt examples: Guided examples of how Bolt can help you automate common tasks.</a></li>
+            <li><a href="https://github.com/puppetlabs/bolt-examples">Bolt example repo: A curated repo of common Bolt patterns that you can use to reference syntax and structure</a></li>
```
Comment (hestonhoffman) [inline documentation/bolt.md:58]:
```suggestion
            <li><a class="xref" href="bolt_examples.md">Bolt examples</a> - Guided examples of how Bolt can help you automate common tasks.</li>
            <li><a href="https://github.com/puppetlabs/bolt-examples">Bolt example repo</a> - A curated repo of common Bolt patterns that you can use to reference syntax and structure</li>
```
Just to follow the style of other entries on this page

File: documentation/bolt.ditamap [hunk 4]
```diff
@@ -74,13 +74,22 @@
...
-    <topicref href="bolt_examples.md" format="markdown"/>
     <topichead navtitle="Using Bolt with Puppet">
         <topicref href="bolt_configure_orchestrator.md" format="markdown"/>
         <topicref href="bolt_connect_puppetdb.md" format="markdown"/>
     </topichead>
-    <topicref href="tig_stack.md" format="markdown"/>
+    <topicref href="bolt_examples.md" format="markdown">
+      <topicref href="automating_windows_targets.md" format="markdown">
+        <topicmeta>
+          <shortdesc>Examples of how to run Powershell scripts, use Bolt inventory, convert a script to a task, and write a Bolt plan, all on Windows.
```
Comment (beechtom) [inline documentation/bolt.ditamap:84]:
```suggestion
          <shortdesc>Examples of how to run Powershell scripts, use Bolt inventory, convert a script to a task, and write a Bolt plan, all on Windows.</shortdesc>
```

File: documentation/bolt.md (hunk 3 above)
Comment (beechtom) [inline documentation/bolt.md:58]:
```suggestion
            <li><a href="https://github.com/puppetlabs/bolt-examples">Bolt example repo: A curated repo of common Bolt patterns that you can use to reference syntax and structure.</a></li>
```

**Pull request merged**